lib/MusicBrainz/Server/Controller/Role/Cleanup.pm

```diff
@@ -1,17 +1,35 @@
 package MusicBrainz::Server::Controller::Role::Cleanup;
 use Moose::Role;
+use MusicBrainz::Server::Constants qw( entities_with );
 use MusicBrainz::Server::Data::Utils qw( boolean_to_json );
 use namespace::autoclean;
 
+my %model_to_edit_type = entities_with('add_edit_type', take => sub {
+    my ($type, $properties) = @_;
+    return ($properties->{model} => $properties->{add_edit_type});
+});
+
 after show => sub {
     my ($self, $c) = @_;
     my $entity = $c->stash->{entity};
-    my $eligible_for_cleanup = $c->model( $self->config->{model} )->is_empty($entity->id);
+
+    my $model = $self->config->{model};
+
+    # If the creation edit is the only pending edit
+    # and the entity is otherwise empty,
+    # it will be removed as soon as the creation edit passes
+    my $creation_edit = $c->model('Edit')->find_creation_edit(
+        $model_to_edit_type{$model},
+        $entity->id);
+
+    my $has_no_edits = $entity->edits_pending == 0;
+    my $has_creation_edit_only = $entity->edits_pending == 1 &&
+        $creation_edit && $creation_edit->is_open;
+    my $is_empty = $c->model($model)->is_empty($entity->id);
+    my $eligible_for_cleanup = $is_empty &&
+        ($has_no_edits || $has_creation_edit_only);
```

Comment:
You can avoid calling `is_empty` or `find_creation_edit` entirely depending on the edits pending count. Maybe something like this?
```suggestion
    my $edits_pending = $entity->edits_pending;
    if ($edits_pending == 1) {
        my $creation_edit = $c->model('Edit')->find_creation_edit(
            $model_to_edit_type{$model},
            $entity->id,
        );
        if ($creation_edit && $creation_edit->is_open) {
            $edits_pending = 0;
        }
    }
    my $eligible_for_cleanup =
        $edits_pending == 0 &&
        $c->model($model)->is_empty($entity->id);
```